### ISSUE TITLE
macos dock icon per forwarded application

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#1727]
Assuming that this is even possible (TBC), we would need to generate an application bundle on-the-fly for each application we forward. (and ship png2icns to convert icons - even pillow uses exec to save icns files: https://github.com/python-pillow/Pillow/blob/master/PIL/IcnsImagePlugin.py

Then this process would need to send messages back to the xpra client process to tell it to:
* raise windows when the dock icon is clicked
* close those windows when the dock icon's "quit" option is clicked
* this may help with showing native notifications, see #1305#comment:5
* this may help with forwarding application menus too

We should still keep xpra's own dock icon for accessing settings, etc

**Comment by ricochet1k** [comment]
Looks like you can set the view used to render the dock tile for the Application: https://developer.apple.com/documentation/appkit/nsdocktile/1525995-contentview

**Comment by totaam** [comment]
@ricochet1k that's not what we want. We want multiple docks for a single process and this cannot be done on MacOS, setting the dock's contents is not the issue.

**Comment by ricochet1k** [comment]
Yes, I'm aware. I was referring to the bit in the description talking about png2icns and generating bundles on the fly. Although I guess if you want to generate bundles for every application not currently running so they can be launched?

**Comment by totaam** [comment]
We can't generate bundles for every application not currently running as the list of all applications is never fully known.
Anyway, we can probably get away with re-using the same bundle and change the dock icon quickly after startup.

AFAIK, once we have a separate process, setting the dock icon is the easy part:
https://github.com/Xpra-org/xpra/blob/da232585f6a837b2393bbb18ff812ac028568936/xpra/platform/darwin/tray.py#L76-L78

**Comment by totaam** [comment]
Likely the same solution as #3721
